### PR TITLE
Extension test refactor

### PR DESCRIPTION
**Pull request by OlajumokeFaj**
### WHY are these changes introduced?

[Issue 117](https://github.com/Shopify/shopify-app-cli/issues/1177)
This is part of a bigger task to refactor test code for easy readability

### Update checklist

- [ ] ~~I've added a CHANGELOG entry for this PR (if the change is public-facing)~~
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).

### CODE REVIEW

File: test/project_types/extension/commands/extension_command_test.rb [hunk 1]
```diff
@@ -40,21 +36,16 @@ def test_extension_type_aborts_if_a_lazily_initialized_field_of_an_unknown_type_
         ])
       end
 
-      def test_extension_type_returns_a_lazy_specification_handler
-        setup_temp_project
```
Comment (t6d) [inline test/project_types/extension/commands/extension_command_test.rb:44]: Were you able to verify that removing `setup_temp_project` in this test and other tests below does not break the tests. Arguably, if we can take this code out, that would be great because it is clearly at best executed for its side effects and at worst completely unnecessary. 

In short, just double check that this didn't do anything important and it is indeed safe to remove. 😄

File: test/project_types/extension/commands/register_test.rb [hunk 2]
```diff
@@ -6,14 +6,16 @@ module Extension
...
-      include ExtensionTestHelpers::TempProjectSetup
       include ExtensionTestHelpers::Messages
 
       def setup
         super
         ShopifyCli::ProjectType.load_type(:extension)
-        setup_temp_project(registration_id: nil)
-        @app = Models::App.new(api_key: @api_key, secret: @api_secret)
+        @project = ExtensionTestHelpers.fake_extension_project(with_mocks: true, registration_id: nil)
+        @context = ExtensionTestHelpers.fake_context
```
Comment (t6d) [inline test/project_types/extension/commands/register_test.rb:15]: Let's revert this change and rely on `@context` being setup in our `Minitest::Test` override.

**Review by t6d — APPROVED** [review]

**Review by cecyc — APPROVED** [review]
LG2M! 👍
